packages/wrangler/src/traverse-module-graph.ts

```diff
@@ -0,0 +1,55 @@
+import { readdir } from "node:fs/promises";
+import path from "node:path";
+import chalk from "chalk";
+import { logger } from "./logger";
+import { matchFiles, parseRules } from "./module-collection";
+import type { BundleResult } from "./bundle";
+import type { Config } from "./config";
+import type { Entry } from "./entry";
+
+async function getFiles(root: string): Promise<string[]> {
+	const files = [];
+	for (const file of await readdir(root, { withFileTypes: true })) {
+		if (file.isDirectory()) {
+			files.push(...(await getFiles(path.join(root, file.name))));
+		} else {
+			files.push(path.join(root, file.name));
+		}
+	}
+	return files;
+}
+
+export default async function traverseModuleGraph(
+	entry: Entry,
+	rules: Config["rules"]
+): Promise<BundleResult> {
+	const files = await getFiles(entry.moduleRoot);
+
+	const parsedRules = parseRules(rules);
+	const modules = (
+		await matchFiles(files, parsedRules.rules, parsedRules.rulesToRemove)
+	)
```

Comment:
How about we change `matchFiles()` so that we can call it like:

```suggestion
	const modules = await matchFiles(files, parseRules(rules));
```